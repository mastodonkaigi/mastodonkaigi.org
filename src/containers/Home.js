import React from 'react';
import { FormattedMessage } from 'react-intl';
import styled from 'styled-components';

const Hero = styled.header`
  align-items: center;
  display: flex;
  flex-direction: column;
  min-height: 50vh;
  justify-content: center;
`;

const Title = styled.h1`
  font-size: 2.75rem;
  margin: 0;
  padding: 0;
`;

const Description = styled.p`
  font-size: 0.95rem;
  margin: 0;
  padding: 1rem 0;
`;

export default function Home() {
  return (
    <Hero>
      <Title>
        <FormattedMessage id="home.title" />
      </Title>
      <Description>
        <FormattedMessage id="home.welcome" />
      </Description>
    </Hero>
  );
}
