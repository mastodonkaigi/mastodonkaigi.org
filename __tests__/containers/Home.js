import React from 'react';
import { IntlProvider } from 'react-intl';
import renderer from 'react-test-renderer';
import Home from '../../src/containers/Home';
import messages from '../../src/locales/en';

test('Home container renders correctly', () => {
  const container = renderer.create((
    <IntlProvider locale="en" messages={messages}>
      <Home />
    </IntlProvider>
  ));
  const tree = container.toJSON();
  expect(tree).toMatchSnapshot();
});
