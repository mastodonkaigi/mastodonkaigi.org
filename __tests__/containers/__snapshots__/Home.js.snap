// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home container renders correctly 1`] = `
<header
  className="sc-bdVaJa keiair"
>
  <h1
    className="sc-bwzfXH hekLpX"
  >
    <span>
      MastodonKaigi
    </span>
  </h1>
  <p
    className="sc-htpNat kacuyk"
  >
    <span>
      Welcome to MastodonKaigi's website!
    </span>
  </p>
</header>
`;
